File: Python3/Medium/Accepted/jump-game-iii_0.py
class Solution:
    def canReach(self, arr: List[int], start: int) -> bool:
        
        visited = set()
        queue = collections.deque([start])
        while queue:
            i = queue.popleft()
            if arr[i] == 0: return True
            visited.add(i)
            for j in [i + arr[i], i - arr[i]]:
                if 0 <= j < len(arr) and j not in visited:
                    queue.append(j)
        return False

File: Rust/Medium/Accepted/jump-game-iii_0.rs
impl Solution {
    pub fn can_reach(arr: Vec<i32>, start: i32) -> bool {
        
        let mut visited = vec![false; arr.len()];
        let mut stack = vec![start as usize];
        while let Some(i) = stack.pop() {
            if arr[i] == 0 {
                return true;
            }
            visited[i] = true;
            if i as i32 + arr[i] < arr.len() as i32 && !visited[i + arr[i] as usize] {
                stack.push(i + arr[i] as usize);
            }
            if i as i32 - arr[i] >= 0 && !visited[i - arr[i] as usize] {
                stack.push(i - arr[i] as usize);
            }
        }
        false
    }
}